src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -792,7 +790,8 @@ private unsafe void DisposeInternal()
             if (lockTaken == false)
             {
                 ForTestingPurposes?.DisposeLog?.Invoke(Name, "Acquiring cluster lock");
-                Monitor.Enter(_clusterLocker);
+                // ReSharper disable once MethodSupportsCancellation
+                _clusterLocker.Wait();
```

Comment:
What would free this on cluster shutdown?

Reply:
Why do we have to wait in a syncronous fashion here?

Reply:
We only care about the last published index, no? Can't we check and free early here?

Reply:
This is dispose of document database, sync method, have you seen the comment for this code 'must acquire the lock in order to prevent concurrent access to index files' ?

Reply:
This is called on disposal of the database to prevent concurrent access to index files.